onnxruntime/core/optimizer/constant_folding.cc

```diff
@@ -174,6 +174,10 @@ Status ConstantFolding::ApplyImpl(Graph& graph, bool& modified, int graph_level,
     }
 
     if (converted_to_constant) {
+      // Remove single-output node chain for inputs of the node
+      for (auto p_ip_node = node->InputNodesBegin(); p_ip_node != node->InputNodesEnd(); ++p_ip_node) {
+        graph_utils::RemoveNodesWithOneOutputBottomUp(graph, *p_ip_node);
```

Comment:
I wonder how this works... 
From the code, it looks like it's deleting the input nodes of current node. 
How would it trace up to the chain and delete them in a row? 
Did I misunderstand something here?

Reply:
synced offline. 
RemoveNodesWithOneOutputBottomUp takes care of the node removal in a chain

Reply:
the function graph_utils::RemoveNodesWithOneOutputBottomUp() takes care of deleting the chain of unused inputs